docs/content/docs/deep/componets.md

```diff
@@ -0,0 +1,113 @@
+
+## Data Objects and Distributed Data Structures (DDSes)
+
+Each container gets a `ContainerSchema`, which defines a tree (currently implemented using shared directory) of statically defined DDSes and DataObjects.
+The container also can contain dynamically allocated DDSes and DataObjects which can be referenced from others via an `IFluidHandle`, and garbage collected when unreferenced.
+
+When configuring the container, the application specifies which DDSes and DataObjects implementations the container supports, which implicitly selects the concrete implementation (which package version and which copy of the package if there are multiple) the container will use for the DDses and DataObjects.
+
+Smaller libraries/components which the applications uses to work with data within the container should define version ranges of the Fluid client libraries they are compatible with, and the application should ensure all code using the same container is using the same copies of the Fluid client packages.
+Some exceptions to this are known to work: TODO: list working/supported exceptions here.
+
+An application can select which DDses (TODO: or DataObjects?) in the container to load, allowing both reading and writing to containers which are only partially loaded.
+(TODO: note scalability implications due to summary client, op bandwidth).
+
+Within a container, all DDses are synchronized, and operations can between different DDSes are ordered.
+Each DDS has its own merge resolution, so there is no way to do things like multi DDS transactions, or have changes in one DDS impact how a change is merged in another, which can limit the fidelity of operations like moves of content between DDSes, even within a container.
+
+The Fluid Framework provides a garbage collector which can delete DDSes when they become unreferenced by the container (no reachable from the root DDS, which is typically the shared directory created from the container schema).
+
+### Data Objects
+
+Data objects are an under development abstraction on-top of DDSes (see below) which can allow reusable logic build on DDSes to be easily reused.
+
+Currently no public APIs exist for defining new data-object types, but the existing experimental ones can be used in container schema.
+
+### Distributed Data Structures: DDSes
+
+A data structure in a container, with its own self contained editing and merge resolution logic.
+May contain handles to other DDSes.
+
+Some DDSes (mainly SHaredTree) provide their own functionality for further componentization of their content.
+
+While while the user of the container can choose which DDSes to load, DDSes themselves currently can not support being only partially loaded: either the whole DDS is available or not.
+The one exception to this is blobs which the user of the container can asynchronously upload then insert handle to them into the DDS.
```

Comment:
I'd probably remove reference to blobs. They are not implemented for AFR, not part of public API surface, and it's not really an exception to partial DDS loading (I'd see a connection if there was a way to load a subset of blob).